HADOOP-18401. No ARM binaries in branch-3.3.x releases.

steveloughran opened the pull request:
### Description of PR

This change disables the failing RUN command from the docker script and simply installs python -which the yarn ui needs.

### How was this patch tested?

trying to build locally on a docker container under macos.

hard to tell if the node.js build has hung or is just really, really slow. never tried it before outside a release build, and that I delegate to remote systems

### For code changes:

- [X] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

steveloughran committed:
HADOOP-18401. No ARM binaries in branch-3.3.x releases.

This change disables the failing RUN command from the docker
script and simply installs python -which the yarn ui needs.

Change-Id: I9fc89850633d680c528abea93d6ee79e32f8f5cd

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/1/console in case of problems.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  10m 26s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 33s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  28m 27s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  hadolint  |   0m  2s | [/results-hadolint.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/1/artifact/out/results-hadolint.txt) |  The patch generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0)  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  24m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   |  82m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs |
| uname | Linux 3afcdf76824f 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 3f5e2e5b7f3cc5dc82c8c36d573bd518372f236f |
| Max. process+thread count | 527 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/1/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

iwasakims commented on dev-support/docker/Dockerfile_aarch64:
```diff
@@ -130,7 +130,10 @@ ENV PATH "${PATH}:/opt/protobuf/bin"
-RUN pip3 install pylint==2.6.0 python-dateutil==2.8.1
+# Disabled due to dependency availability/build issues (see HADOOP-18401)
+# instead "python" is added so the yarn-ui will build)
+# RUN pip3 install pylint==2.6.0 python-dateutil==2.8.1
```
`mvn site` failed without this. @steveloughran 

```
$ ./dev-support/bin/create-release --docker --dockercache
...
ERROR:root:This script requires python-dateutil module to be installed. You can install it using:
         pip install python-dateutil
```

Assuming Python 3.6, `lazy-object-proxy==1.6.0` worked for me to fix the `SyntaxError: future feature annotations is not defined`.

```
RUN pip3 install pylint==2.6.0 python-dateutil==2.8.1 lazy-object-proxy==1.6.0
```

steveloughran replied:
using ayush's patch; rolling back this change. thanks for the feedback tho'

steveloughran committed:
HADOOP-18401. No ARM binaries in branch-3.3.x releases.

applied patch by Ayush Saxena to sync with x86 dockerfile

Change-Id: I6cfa869a0d923e5f2141af7bd76c01ed16a3c388

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/2/console in case of problems.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   6m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  24m 41s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  23m  8s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   |  73m 19s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs |
| uname | Linux 751d170885a5 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 0a49b7902aadaee5368605d8496276a264ab260b |
| Max. process+thread count | 550 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/2/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
brought up container and did build from inside it
```
mvn install -Pdist -Pnative -Pyarn-ui -DskipTests -Dmaven.javadoc.skip=true
mvn package -Pdist -Pnative -Pyarn-ui -DskipTests -Dmaven.javadoc.skip=true
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Hadoop Main 3.3.9-SNAPSHOT:
[INFO]
[INFO] Apache Hadoop Main ................................. SUCCESS [  1.494 s]
[INFO] Apache Hadoop Build Tools .......................... SUCCESS [  7.522 s]
[INFO] Apache Hadoop Project POM .......................... SUCCESS [  1.002 s]
[INFO] Apache Hadoop Annotations .......................... SUCCESS [  0.709 s]
[INFO] Apache Hadoop Assemblies ........................... SUCCESS [  0.108 s]
[INFO] Apache Hadoop Project Dist POM ..................... SUCCESS [  0.718 s]
[INFO] Apache Hadoop Maven Plugins ........................ SUCCESS [  1.799 s]
[INFO] Apache Hadoop MiniKDC .............................. SUCCESS [  0.489 s]
[INFO] Apache Hadoop Auth ................................. SUCCESS [  2.939 s]
[INFO] Apache Hadoop Auth Examples ........................ SUCCESS [  1.306 s]
[INFO] Apache Hadoop Common ............................... SUCCESS [ 35.524 s]
[INFO] Apache Hadoop NFS .................................. SUCCESS [  2.033 s]
[INFO] Apache Hadoop KMS .................................. SUCCESS [  2.258 s]
[INFO] Apache Hadoop Registry ............................. SUCCESS [  2.670 s]
[INFO] Apache Hadoop Common Project ....................... SUCCESS [  0.037 s]
[INFO] Apache Hadoop HDFS Client .......................... SUCCESS [ 14.874 s]
[INFO] Apache Hadoop HDFS ................................. SUCCESS [ 32.769 s]
[INFO] Apache Hadoop HDFS Native Client ................... SUCCESS [01:33 min]
[INFO] Apache Hadoop HttpFS ............................... SUCCESS [  3.492 s]
[INFO] Apache Hadoop HDFS-NFS ............................. SUCCESS [  1.889 s]
[INFO] Apache Hadoop HDFS-RBF ............................. SUCCESS [  7.042 s]
[INFO] Apache Hadoop HDFS Project ......................... SUCCESS [  0.059 s]
[INFO] Apache Hadoop YARN ................................. SUCCESS [  0.034 s]
[INFO] Apache Hadoop YARN API ............................. SUCCESS [  8.773 s]
[INFO] Apache Hadoop YARN Common .......................... SUCCESS [ 11.154 s]
[INFO] Apache Hadoop YARN Server .......................... SUCCESS [  0.036 s]
[INFO] Apache Hadoop YARN Server Common ................... SUCCESS [  6.330 s]
[INFO] Apache Hadoop YARN NodeManager ..................... SUCCESS [  9.498 s]
[INFO] Apache Hadoop YARN Web Proxy ....................... SUCCESS [  1.596 s]
[INFO] Apache Hadoop YARN ApplicationHistoryService ....... SUCCESS [  2.825 s]
[INFO] Apache Hadoop YARN Timeline Service ................ SUCCESS [  2.141 s]
[INFO] Apache Hadoop YARN ResourceManager ................. SUCCESS [ 15.877 s]
[INFO] Apache Hadoop YARN Server Tests .................... SUCCESS [  1.853 s]
[INFO] Apache Hadoop YARN Client .......................... SUCCESS [  3.065 s]
[INFO] Apache Hadoop YARN SharedCacheManager .............. SUCCESS [  1.474 s]
[INFO] Apache Hadoop YARN Timeline Plugin Storage ......... SUCCESS [  1.430 s]
[INFO] Apache Hadoop YARN TimelineService HBase Backend ... SUCCESS [  0.032 s]
[INFO] Apache Hadoop YARN TimelineService HBase Common .... SUCCESS [  1.976 s]
[INFO] Apache Hadoop YARN TimelineService HBase Client .... SUCCESS [  2.650 s]
[INFO] Apache Hadoop YARN TimelineService HBase Servers ... SUCCESS [  0.034 s]
[INFO] Apache Hadoop YARN TimelineService HBase Server 1.2  SUCCESS [  3.514 s]
[INFO] Apache Hadoop YARN TimelineService HBase tests ..... SUCCESS [  2.564 s]
[INFO] Apache Hadoop YARN Router .......................... SUCCESS [  2.041 s]
[INFO] Apache Hadoop YARN TimelineService DocumentStore ... SUCCESS [  1.599 s]
[INFO] Apache Hadoop YARN Applications .................... SUCCESS [  0.032 s]
[INFO] Apache Hadoop YARN DistributedShell ................ SUCCESS [  1.656 s]
[INFO] Apache Hadoop YARN Unmanaged Am Launcher ........... SUCCESS [  1.192 s]
[INFO] Apache Hadoop MapReduce Client ..................... SUCCESS [  0.549 s]
[INFO] Apache Hadoop MapReduce Core ....................... SUCCESS [  6.982 s]
[INFO] Apache Hadoop MapReduce Common ..................... SUCCESS [  3.507 s]
[INFO] Apache Hadoop MapReduce Shuffle .................... SUCCESS [  2.075 s]
[INFO] Apache Hadoop MapReduce App ........................ SUCCESS [  4.769 s]
[INFO] Apache Hadoop MapReduce HistoryServer .............. SUCCESS [  2.947 s]
[INFO] Apache Hadoop MapReduce JobClient .................. SUCCESS [  5.496 s]
[INFO] Apache Hadoop Mini-Cluster ......................... SUCCESS [  1.389 s]
[INFO] Apache Hadoop YARN Services ........................ SUCCESS [  0.031 s]
[INFO] Apache Hadoop YARN Services Core ................... SUCCESS [  4.177 s]
[INFO] Apache Hadoop YARN Services API .................... SUCCESS [  2.397 s]
[INFO] Apache Hadoop YARN Application Catalog ............. SUCCESS [  0.033 s]
[INFO] Apache Hadoop YARN Application Catalog Webapp ...... SUCCESS [  8.285 s]
[INFO] Apache Hadoop YARN Application Catalog Docker Image  SUCCESS [  0.058 s]
[INFO] Apache Hadoop YARN Application MaWo ................ SUCCESS [  0.030 s]
[INFO] Apache Hadoop YARN Application MaWo Core ........... SUCCESS [  1.092 s]
[INFO] Apache Hadoop YARN Site ............................ SUCCESS [  0.049 s]
[INFO] Apache Hadoop YARN Registry ........................ SUCCESS [  0.599 s]
[INFO] Apache Hadoop YARN UI .............................. SUCCESS [01:23 min]
[INFO] Apache Hadoop YARN CSI ............................. SUCCESS [  4.619 s]
[INFO] Apache Hadoop YARN Project ......................... SUCCESS [ 22.600 s]
[INFO] Apache Hadoop MapReduce HistoryServer Plugins ...... SUCCESS [  1.060 s]
[INFO] Apache Hadoop MapReduce NativeTask ................. SUCCESS [  8.127 s]
[INFO] Apache Hadoop MapReduce Uploader ................... SUCCESS [  1.196 s]
[INFO] Apache Hadoop MapReduce Examples ................... SUCCESS [  2.067 s]
[INFO] Apache Hadoop MapReduce ............................ SUCCESS [  4.184 s]
[INFO] Apache Hadoop MapReduce Streaming .................. SUCCESS [  2.240 s]
[INFO] Apache Hadoop Distributed Copy ..................... SUCCESS [  2.559 s]
[INFO] Apache Hadoop Client Aggregator .................... SUCCESS [  2.440 s]
[INFO] Apache Hadoop Dynamometer Workload Simulator ....... SUCCESS [  1.592 s]
[INFO] Apache Hadoop Dynamometer Cluster Simulator ........ SUCCESS [  2.511 s]
[INFO] Apache Hadoop Dynamometer Block Listing Generator .. SUCCESS [  1.442 s]
[INFO] Apache Hadoop Dynamometer Dist ..................... SUCCESS [  4.362 s]
[INFO] Apache Hadoop Dynamometer .......................... SUCCESS [  0.033 s]
[INFO] Apache Hadoop Archives ............................. SUCCESS [  1.349 s]
[INFO] Apache Hadoop Archive Logs ......................... SUCCESS [  1.555 s]
[INFO] Apache Hadoop Rumen ................................ SUCCESS [  2.493 s]
[INFO] Apache Hadoop Gridmix .............................. SUCCESS [  2.320 s]
[INFO] Apache Hadoop Data Join ............................ SUCCESS [  1.539 s]
[INFO] Apache Hadoop Extras ............................... SUCCESS [  1.662 s]
[INFO] Apache Hadoop Pipes ................................ SUCCESS [  0.385 s]
[INFO] Apache Hadoop OpenStack support .................... SUCCESS [  1.543 s]
[INFO] Apache Hadoop Amazon Web Services support .......... SUCCESS [  5.894 s]
[INFO] Apache Hadoop Kafka Library support ................ SUCCESS [  1.027 s]
[INFO] Apache Hadoop Azure support ........................ SUCCESS [  5.207 s]
[INFO] Apache Hadoop Aliyun OSS support ................... SUCCESS [  1.213 s]
[INFO] Apache Hadoop Scheduler Load Simulator ............. SUCCESS [  2.062 s]
[INFO] Apache Hadoop Resource Estimator Service ........... SUCCESS [  1.763 s]
[INFO] Apache Hadoop Azure Data Lake support .............. SUCCESS [  1.270 s]
[INFO] Apache Hadoop Image Generation Tool ................ SUCCESS [  1.325 s]
[INFO] Apache Hadoop Tools Dist ........................... SUCCESS [  3.302 s]
[INFO] Apache Hadoop Common Benchmark ..................... SUCCESS [02:41 min]
[INFO] Apache Hadoop Tools ................................ SUCCESS [  0.079 s]
[INFO] Apache Hadoop Client API ........................... SUCCESS [01:01 min]
[INFO] Apache Hadoop Client Runtime ....................... SUCCESS [ 56.989 s]
[INFO] Apache Hadoop Client Packaging Invariants .......... SUCCESS [  0.182 s]
[INFO] Apache Hadoop Client Test Minicluster .............. SUCCESS [01:32 min]
[INFO] Apache Hadoop Client Packaging Invariants for Test . SUCCESS [  0.121 s]
[INFO] Apache Hadoop Client Packaging Integration Tests ... SUCCESS [  0.096 s]
[INFO] Apache Hadoop Distribution ......................... SUCCESS [01:25 min]
[INFO] Apache Hadoop Client Modules ....................... SUCCESS [  0.058 s]
[INFO] Apache Hadoop Tencent COS Support .................. SUCCESS [  1.381 s]
[INFO] Apache Hadoop Cloud Storage ........................ SUCCESS [  0.532 s]
[INFO] Apache Hadoop Cloud Storage Project ................ SUCCESS [  0.027 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16:24 min
[INFO] Finished at: 2022-09-30T13:59:45Z
[INFO] ------------------------------------------------------------------------

stevel@4ed2b9e0c7da:~/hadoop$ ls -l hadoop-dist/target/
total 691928
drwxr-xr-x  3 stevel users        96 Sep 29 16:50 antrun
drwxr-xr-x  3 stevel users        96 Sep 29 16:50 classes
drwxr-xr-x 15 stevel users       480 Sep 30 13:58 hadoop-3.3.9-SNAPSHOT
-rw-r--r--  1 stevel users  36391566 Sep 29 16:52 hadoop-3.3.9-SNAPSHOT-src.tar.gz
-rw-r--r--  1 stevel users 666689305 Sep 30 13:59 hadoop-3.3.9-SNAPSHOT.tar.gz
drwxr-xr-x  3 stevel users        96 Sep 29 16:43 hadoop-tools-deps
drwxr-xr-x  3 stevel users        96 Sep 29 16:50 maven-shared-archive-resources
-rw-r--r--  1 stevel users       698 Sep 29 16:17 rat.txt
drwxr-xr-x  3 stevel users        96 Sep 29 16:50 test-classes
drwxr-xr-x  2 stevel users        64 Sep 29 16:50 test-dir
```

that should be enough to get the native libs built up in the tarball, which is the only thing we'd want to sign and publish.

now trying `dev-support/bin/create-release --docker`

that does not work, still trying to use the old file. any ideas?
```
dev-support/bin/create-release --docker
[+] Building 12.8s (14/21)
 => [internal] load build definition from Dockerfile                                                                       0.0s
 => => transferring dockerfile: 5.99kB                                                                                     0.0s
 => [internal] load .dockerignore                                                                                          0.0s
 => => transferring context: 2B                                                                                            0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                           0.6s
 => [ 1/18] FROM docker.io/library/ubuntu:bionic@sha256:6fec50623d6d37b7f3c14c5b6fc36c73fd04aa8173d59d54dba00da0e7ac50ee   0.0s
 => CACHED [ 2/18] WORKDIR /root                                                                                           0.0s
 => CACHED [ 3/18] RUN echo APT::Install-Recommends "0"; > /etc/apt/apt.conf.d/10disableextras                             0.0s
 => CACHED [ 4/18] RUN echo APT::Install-Suggests "0"; >>  /etc/apt/apt.conf.d/10disableextras                             0.0s
 => CACHED [ 5/18] RUN apt-get -q update     && apt-get -q install -y --no-install-recommends         apt-utils         b  0.0s
 => CACHED [ 6/18] RUN locale-gen en_US.UTF-8                                                                              0.0s
 => CACHED [ 7/18] RUN mkdir -p /opt/spotbugs     && curl -L -s -S https://github.com/spotbugs/spotbugs/releases/download  0.0s
 => CACHED [ 8/18] RUN apt-get -q update     && apt-get -q install -y --no-install-recommends openjdk-8-jdk libbcprov-jav  0.0s
 => CACHED [ 9/18] RUN mkdir -p /opt/protobuf-src     && curl -L -s -S       https://github.com/protocolbuffers/protobuf/  0.0s
 => CACHED [10/18] RUN apt-get -q update     && apt-get -q install -y --no-install-recommends maven     && apt-get clean   0.0s
 => CANCELED [11/18] RUN pip3 install pylint==2.6.0 python-dateutil==2.8.1                                                12.1s^C^C^C
context canceled
```

steveloughran committed:
HADOOP-18401. No ARM binaries in branch-3.3.x releases.

create-release uses aarch64 image when machine type is arm64-apple-darwin21
also prints this fact out...

Change-Id: I01bd359c1f81ce632eaf527f8c644b5d85865532

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/3/console in case of problems.

steveloughran committed:
HADOOP-18401. lets get bash right

(doing the releases in a separate dir from editing, so pushing via github)

Change-Id: Ib26a3cbb4d785526a46725bcf38cea61d8916e4d

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/4/console in case of problems.

steveloughran commented:
...last patch switches to the right dockerfile, doesn't work though...even though the devenv does
```
cd /build/source
/usr/bin/mvn -Dmaven.repo.local=/home/stevel/.m2/repository -pl hadoop-maven-plugins -am clean install
The JAVA_HOME environment variable is not defined correctly
This environment variable is needed to run this program
NB: JAVA_HOME should point to a JDK not a JRE
```

iwasakims commented on dev-support/docker/Dockerfile_aarch64:
```diff
@@ -17,7 +17,7 @@
 # See BUILDING.txt.
 
-FROM ubuntu:bionic
+FROM ubuntu:focal
```
Upgrading from bionic to focal in a point release sounds aggressive. Dockerfile (for x86_64) still uses bionic in branch-3.3.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 34s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  15m  3s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  24m 37s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  2s |  |  No new issues.  |
| -1 :x: |  shellcheck  |   0m  1s | [/results-shellcheck.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/3/artifact/out/results-shellcheck.txt) |  The patch generated 4 new + 0 unchanged - 0 fixed = 4 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  23m  7s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 49s |  |  The patch does not generate ASF License warnings.  |
|  |   |  66m 37s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint |
| uname | Linux 202d7a40d0be 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 13206b70a1aed831fbc7b30f0f2a3fd727bb3cc9 |
| Max. process+thread count | 726 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/3/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 53s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 51s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  26m 29s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  24m 47s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   |  70m  3s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint |
| uname | Linux b7a431e09c49 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / e2962cff9064a21564bd08b94de2a44b9b40b936 |
| Max. process+thread count | 522 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/4/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn commented:
Java Home isn't getting set correctly in the release script.
https://github.com/steveloughran/hadoop/blob/build/HADOOP-18401-arm-64-branch-3.3/dev-support/bin/create-release#L517

I was able to repro the issue, changed the condition to include arm64 and then had to do a docker system prune -a, else this statement wasn't getting picked up, post that it worked. Do give a check if the statement is getting executed if you get the same error

```
diff --git a/dev-support/bin/create-release b/dev-support/bin/create-release
index 73af023390e..a93dc77fba9 100755
--- a/dev-support/bin/create-release
+++ b/dev-support/bin/create-release
@@ -514,7 +514,8 @@ function dockermode
 
     # we always force build with the OpenJDK JDK
     # but with the correct version
-    if [ "$CPU_ARCH" = "aarch64" ]; then
+
+    if [[ "$CPU_ARCH" = "aarch64"  || "$CPU_ARCH" = "arm64" ]]; then
       echo "ENV JAVA_HOME /usr/lib/jvm/java-${JVM_VERSION}-openjdk-arm64"
     else
       echo "ENV JAVA_HOME /usr/lib/jvm/java-${JVM_VERSION}-openjdk-amd64"
```

steveloughran committed:
HADOOP-18401. java home on m1

Change-Id: I25c5cc9037962b31e1180a87eeccd7fc6e37a248

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/5/console in case of problems.

steveloughran committed:
HADOOP-18401. java home on m1

Change-Id: Ida29b0da3b9166b772d38f256aa79628c40aa4df

steveloughran commented:
thanks; trying again

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/6/console in case of problems.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   4m 23s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 54s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  25m 41s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  22m 54s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 43s |  |  The patch does not generate ASF License warnings.  |
|  |   |  71m 27s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint |
| uname | Linux 08d10ac82f1b 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / f071cec78621362d77f37b00b3f2f967f7162294 |
| Max. process+thread count | 643 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/6/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  10m 44s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 49s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  26m 15s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  24m 40s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   |  79m 45s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint |
| uname | Linux 1ea8ce343fc5 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / f071cec78621362d77f37b00b3f2f967f7162294 |
| Max. process+thread count | 602 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/5/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
status. native tarball is happy but site build fails, and it's related to something patching MetricsSystem.java

```
****************************************************************************
                          Maven Build and Install
****************************************************************************


$ /usr/bin/mvn -Dmaven.repo.local=/home/stevel/.m2/repository install -Pdist,src,yarn-ui -DskipTests -Dtar -Pnative -Drequire.snappy -Drequire.openssl -Drequire.fuse > /build/source/patchprocess/mvn_install.log 2>&1
$ cd /build/source
$ mv /build/source/hadoop-dist/target/hadoop-3.3.9-SNAPSHOT.tar.gz /build/source/target/artifacts/hadoop-3.3.9-SNAPSHOT.tar.gz
$ mv /build/source/hadoop-dist/target/hadoop-3.3.9-SNAPSHOT-src.tar.gz /build/source/target/artifacts/hadoop-3.3.9-SNAPSHOT-src.tar.gz


****************************************************************************
                                Maven Site
****************************************************************************


$ /usr/bin/mvn -Dmaven.repo.local=/home/stevel/.m2/repository install site site:stage -DskipTests -DskipShade -Pdist,src -Preleasedocs,docs > /build/source/patchprocess/mvn_site.log 2>&1

Failed!
```

from the output
```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.11.0:site (default-site) on project hadoop-common: failed to get report for org.apache.maven.plugins:maven-dependency-plugin: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project hadoop-common: Compilation failure
[ERROR] /build/source/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java:[67,25] method <T>register(java.lang.String,java.lang.String,T) is already defined in class org.apache.hadoop.metrics2.MetricsSystem
[ERROR]
```

git changes
```
!) git status
On branch build/HADOOP-18401-arm-64-branch-3.3
Your branch is up to date with 'github/build/HADOOP-18401-arm-64-branch-3.3'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	hadoop-common-project/hadoop-common/src/site/markdown/UnixShellAPI.md
	hadoop-common-project/hadoop-common/src/site/markdown/release/README.md
	hadoop-common-project/hadoop-common/src/site/markdown/release/index.md
	hadoop-common-project/hadoop-common/src/site/resources/configuration.xsl
	hadoop-common-project/hadoop-common/src/site/resources/core-default.xml
```

it looks like something patching the file, so on that second run it fails, as it is adding what is already there

```
git diff HEAD hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
index fef8c4b7e4b..7b21cc302d1 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
@@ -42,6 +42,18 @@ public abstract class MetricsSystem implements MetricsSystemMXBean {
   @InterfaceAudience.Private
   public abstract MetricsSystem init(String prefix);

+  /**
+   * Register a metrics source
+   * @param <T>   the actual type of the source object
+   * @param source object to register
+   * @param name  of the source. Must be unique or null (then extracted from
+   *              the annotations of the source object.)
+   * @param desc  the description of the source (or null. See above.)
+   * @return the source object
+   * @exception MetricsException
+   */
+  public abstract <T> T register(String name, String desc, T source);
+
   /**
    * Register a metrics source
    * @param <T>   the actual type of the source object
```

ayushtkn commented:
Was quite strange on first look, like why a release script is changing the source code, But Jdiff has a bug, because of which during site generation there is a patching logic, the patch now doesn't apply due to conflicts. This problem should come in x86 also.

Just fix the patch something like below & give a check it can be applied without conflicts.
```
diff --git a/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch b/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch
index 2bd7b63f017..26eb53a990b 100644
--- a/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch
+++ b/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch
@@ -1,5 +1,5 @@
 diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
-index a277abd6e13..1d131d5db6e 100644
+index fef8c4b7e4b..8d2b9339706 100644
 --- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
 +++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
 @@ -42,18 +42,6 @@
@@ -14,7 +14,7 @@ index a277abd6e13..1d131d5db6e 100644
 -   *              the annotations of the source object.)
 -   * @param desc  the description of the source (or null. See above.)
 -   * @return the source object
--   * @exception MetricsException
+-   * @exception MetricsException Metrics Exception.
 -   */
 -  public abstract <T> T register(String name, String desc, T source);
 -
@@ -38,7 +38,7 @@ index a277abd6e13..1d131d5db6e 100644
 +   *              the annotations of the source object.)
 +   * @param desc  the description of the source (or null. See above.)
 +   * @return the source object
-    * @exception MetricsException
+    * @exception MetricsException Metrics Exception.
     */
 -  public abstract <T extends MetricsSink>
 -  T register(String name, String desc, T sink);
```

In case interested, this patching happens here:
https://github.com/steveloughran/hadoop/blob/build/HADOOP-18401-arm-64-branch-3.3/hadoop-project-dist/pom.xml#L222

I have triggered the script, will update the result here

steveloughran committed:
HADOOP-18401. fix jdiff patch problem

Change-Id: Ic63dce872e35614d9283f8d65d0930f8faf571b0

hadoop-yetus commented:
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/7/console in case of problems.

steveloughran commented:
let me try it too

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ branch-3.3 Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  32m 14s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  23m 56s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   |  59m 36s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4953 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint |
| uname | Linux 5c1e5471201d 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 2141e9d77b42d824fd3f8989662a7ee9b964d01d |
| Max. process+thread count | 626 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4953/7/console |
| versions | git=2.17.1 maven=3.6.0 hadolint=1.11.1-0-g0e692dd shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn commented:
For me everything worked apart from the last step which moves the changelog because I didn’t change the version and there was no jira marked to 3.3.9-SNAPSHOT(obviously), so should have changed version to 3.3.5 to make it pass the last step.

steveloughran commented:
yeah, i hit the same thing
```
$ mv /build/source/target/hadoop-site-3.3.9-SNAPSHOT.tar.gz /build/source/target/artifacts/hadoop-3.3.9-SNAPSHOT-site.tar.gz
$ cp -p /build/source/hadoop-common-project/hadoop-common/src/site/markdown/release/3.3.9-SNAPSHOT/CHANGELOG*.md /build/source/target/artifacts/CHANGELOG.md
cp: cannot stat '/build/source/hadoop-common-project/hadoop-common/src/site/markdown/release/3.3.9-SNAPSHOT/CHANGELOG*.md': No such file or directory
```

I think this is good to get into -3.3 and then -3.3.5; then we can tune the releases there

steveloughran commented:
merged in; now testing the cherrypick to 3.3.5
